ReactAndroid/src/main/java/com/facebook/react/common/TestIdUtil.java

```diff
@@ -0,0 +1,65 @@
+package com.facebook.react.common;
+
+import android.view.View;
+
+import com.facebook.react.common.annotations.VisibleForTesting;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Utility methods for managing testIDs.
+ */
+public class TestIdUtil {
+    private static final ConcurrentHashMap<String, Integer> TEST_IDS = new ConcurrentHashMap<>();
+    private static final int STARTING_INTERNAL_ID = 10;
+    private static final AtomicInteger INTERNAL_ID = new AtomicInteger(STARTING_INTERNAL_ID);
+
+    /**
+     * Looks for defined resource IDs in R.class by the name of testId and if a matching resource ID is
+     * found it is passed to the view's setId method.  If the given testId cannot be found in R.class,
+     * an internal increment value is assigned instead.
+     *
+     * @param view
+     * @param testId
+     * @param <T>
+     */
+    public static <T extends View> void setTestId(T view, String testId) {
+        int mappedTestId;
+        if (!TEST_IDS.containsKey(testId)) {
+            mappedTestId = view.getResources().getIdentifier(testId, "id", view.getContext().getPackageName());
+            if (mappedTestId <= 0) {
```

Comment:
Does this mean "not found"? If yes add a boolean found to make it easily readable?